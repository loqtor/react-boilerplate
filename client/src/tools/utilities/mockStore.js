export const mockStore = state => ({
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({ ...state }),
});
