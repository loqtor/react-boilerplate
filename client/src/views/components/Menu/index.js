import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { FormattedMessage, injectIntl } from 'react-intl';
import { withRouter, Link } from 'react-router-dom';

const Menu = (props) => {
  const { isOpen } = props;
  const menuClassName = classnames('Menu Menu--sitewide', { 'is-open': isOpen });

  return (
    <nav className={menuClassName} aria-hidden={!isOpen}>
      <h2 className="u-hiddenVisually">
        <FormattedMessage id="MENU" />
      </h2>

      <ul className="Menu-list">
        <li className="Menu-list-item">
          <Link to="/shows/10">Menu item</Link>
        </li>
        <li className="Menu-list-item">
          <Link to="/shows/10">Menu item</Link>
        </li>
        <li className="Menu-list-item">
          <Link to="/shows/10">Menu item</Link>
        </li>
      </ul>

      <div className="Hamburger">
        <span />
        <span />
        <span />
        <span />
      </div>
    </nav>
  );
};

Menu.propTypes = {
  history: PropTypes.shape({}),
  isOpen: PropTypes.bool,
  location: PropTypes.shape({}).isRequired,
};

Menu.defaultProps = {
  history: {},
  isOpen: false,
};

const MenuWithRouter = withRouter(injectIntl(Menu));

export { MenuWithRouter as Menu };
