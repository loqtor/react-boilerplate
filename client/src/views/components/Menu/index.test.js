import React from 'react';
import { shallow } from 'enzyme';
import axe from 'axe-core';
import toJson from 'enzyme-to-json';
import { StaticRouter } from 'react-router-dom';

import shallowToDoc from '../../../tools/utilities/mountToDoc';
import mockStore from '../../../tools/utilities/mockStore';

import Menu from './index';

const setup = (render, props) => {
  const handleLogoutSpy = jest.fn();
  const store = mockStore;

  const defaultProps = {
    isOpen: true,
    logOut: handleLogoutSpy,
  };

  const component = render(
    <StaticRouter store={store}>
      <Menu {...defaultProps} {...props} />
    </StaticRouter>,
  );

  return {
    actual: component,
    handleLogoutSpy,
  };
};

describe('<Menu />', () => {
  it('renders correctly', () => {
    const { actual } = setup(shallow, {});

    expect(toJson(actual)).toMatchSnapshot();
  });

  it('has no accessibility violations', (done) => {
    const { actual } = setup(shallowToDoc, {});
    const componentNode = actual.getDOMNode();

    axe.run(componentNode, (err, { violations }) => {
      expect(err).toBe(null);
      expect(violations).toHaveLength(0);
      done();
    });
  });
});
