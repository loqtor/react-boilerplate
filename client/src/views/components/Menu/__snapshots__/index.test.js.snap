// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(InjectIntl(Menu))
    isOpen={true}
    logOut={[MockFunction]}
  />
</Router>
`;
